packages/amplify-category-auth/src/__tests__/provider-utils/awscloudformation/handlers/resource-handlers.test.ts

```diff
@@ -0,0 +1,72 @@
+
+jest.mock('../../../../provider-utils/awscloudformation/utils/synthesize-resources');
+jest.mock('../../../../provider-utils/awscloudformation/utils/auth-defaults-appliers');
+jest.mock('../../../../provider-utils/supported-services');
+jest.mock('../../../../provider-utils/awscloudformation/auth-inputs-manager/auth-input-state');
+jest.mock('../../../../provider-utils/awscloudformation/utils/generate-auth-stack-template');
+jest.mock('../../../../provider-utils/awscloudformation/utils/message-printer');
+// eslint-disable-next-line spellcheck/spell-checker
+jest.mock('../../../../provider-utils/awscloudformation/utils/amplify-meta-updaters');
+jest.mock('../../../../provider-utils/awscloudformation/utils/auth-sms-workflow-helper');
+jest.mock('amplify-cli-core');
+
+const getSupportedServicesMock = getSupportedServices as jest.MockedFunction<typeof getSupportedServices>;
+getSupportedServicesMock.mockReturnValue({
+  test: {
+    defaultValuesFilename: 'test',
+  },
+});
+
+const testConfig = {
+  hostedUIProviderCreds: 'testProviderCreds',
+};
+
+const getUpdateAuthDefaultsApplierMock = getUpdateAuthDefaultsApplier as jest.MockedFunction<typeof getUpdateAuthDefaultsApplier>;
+getUpdateAuthDefaultsApplierMock.mockReturnValue(jest.fn().mockReturnValue({ ...testConfig }));
+
+const stateManagerMock = stateManager as jest.Mocked<typeof stateManager>;
+stateManagerMock.getMeta.mockReturnValue({
+  auth: {},
+});
+
+const getPostUpdateAuthMetaUpdaterMock = getPostUpdateAuthMetaUpdater as jest.MockedFunction<typeof getPostUpdateAuthMetaUpdater>;
+getPostUpdateAuthMetaUpdaterMock.mockReturnValue(jest.fn());
+
+const getPostUpdateAuthMessagePrinterMock = getPostUpdateAuthMessagePrinter as jest.MockedFunction<typeof getPostUpdateAuthMessagePrinter>;
+getPostUpdateAuthMessagePrinterMock.mockReturnValue(jest.fn());
+
+const removeDeprecatedPropsMock = removeDeprecatedProps as jest.MockedFunction<typeof removeDeprecatedProps>;
+removeDeprecatedPropsMock.mockImplementation(input => input);
+
+const AuthInputStateMock = AuthInputState as jest.MockedClass<typeof AuthInputState>;
+const saveCLIInputPayloadMock = jest.fn();
+AuthInputStateMock.mockImplementation(() => ({
+  saveCLIInputPayload: saveCLIInputPayloadMock,
+} as unknown as AuthInputState));
+
+describe('getUpdateAuthHandler', () => {
+  it('filters cliInputs on env specific params', async () => {
+    const saveParamsFn = jest.fn();
+    const contextStub = {
+      amplify: {
+        saveEnvResourceParameters: saveParamsFn,
+      },
+    } as unknown as $TSContext;
+    const cognitoConfig: CognitoConfiguration = {
+      serviceName: 'test',
+    } as unknown as CognitoConfiguration;
+
+    await getUpdateAuthHandler(contextStub)(cognitoConfig);
+    const actualCliInputsFileContent = saveCLIInputPayloadMock.mock.calls[0][0];
+    expect(Object.keys(actualCliInputsFileContent).includes('hostedUIProviderCreds')).toBe(false);
```

Comment:
Is `hostedUIProviderCreds` the only key that needs to be tested here, or do we need to check that the other `ENV_SPECIFIC_PARAMS` are not present?

Reply:
`hostedUIProviderCreds` is the only one that's critical it not be present but I can update the test to check for the rest as well

Reply:
You could probably just get away with just updating the test name to be more specific if that parameter is the only one we care about in this case. Something to the effect of "filters hostedUIProviderCreds out of cli inputs".

Reply:
I updated the test to check all env specific params